runtime/cudaq/platform/default/rest/helpers/braket/BraketServerHelper.cpp

```diff
@@ -136,7 +146,70 @@ sample_result BraketServerHelper::processResults(ServerMessage &resultsJson,
     }
   }
 
-  return sample_result{ExecutionResult{counts}};
+  // Full execution results include compiler-generated qubits, which are
+  // undesirable to the user.
+  cudaq::ExecutionResult fullExecResults{counts};
+  auto fullSampleResults = cudaq::sample_result{fullExecResults};
+
+  // clang-format off
+  // The following code strips out and reorders the outputs based on output_names.
+  // For example, if `counts` is something like:
+  //      { 11111:62 01111:12 11110:12 01110:12 }
+  // And if we want to discard the first bit (because qubit 0 was a
+  // compiler-generated qubit), that maps to something like this:
+  // -----------------------------------------------------
+  // Qubit  Index - x1234    x1234    x1234    x1234
+  // Result Index - x0123    x0123    x0123    x0123
+  //              { 11111:62 01111:12 11110:12 01110:12 }
+  //              { x1111:62 x1111:12 x1110:12 x1110:12 }
+  //                  \--- v ---/       \--- v ---/
+  //              {    1111:(62+12)     x1110:(12+12)   }
+  //              {    1111:74           1110:24        }
+  // -----------------------------------------------------
+  // clang-format on
+
+  std::vector<ExecutionResult> execResults;
+
+  // Get a reduced list of qubit numbers that were in the original program
+  // so that we can slice the output data and extract the bits that the user
+  // was interested in. Sort by OpenQasm2 qubit number.
+  std::vector<std::size_t> qubitNumbers;
+  qubitNumbers.reserve(output_names.size());
+  for (auto &[result, info] : output_names) {
+    qubitNumbers.push_back(info.qubitNum);
+  }
+
+  // For each original counts entry in the full sample results, reduce it
+  // down to the user component and add to userGlobal. If qubitNumbers is empty,
+  // that means all qubits were measured.
+  if (qubitNumbers.empty()) {
+    execResults.emplace_back(ExecutionResult{fullSampleResults.to_map()});
+  } else {
+    auto subset = fullSampleResults.get_marginal(qubitNumbers);
+    execResults.emplace_back(ExecutionResult{subset.to_map()});
+  }
+
+  // Return a sample result including the global register and all individual
+  // registers.
+  auto ret = cudaq::sample_result(execResults);
```

Comment:
If any `braket` backends have run the mapping pass, you may need changes similar to https://github.com/NVIDIA/cuda-quantum/pull/2328 right here. If we are unsure and under a time crunch, we could perhaps just create an issue and follow up on this later?

Reply:
Created an issue to be addressed later: https://github.com/NVIDIA/cuda-quantum/issues/2425
Thanks for the info!